drivers/spi/spi_mchp_mss.c

```diff
@@ -0,0 +1,496 @@
+		control &= ~MSS_SPI_CONTROL_SPO;
+	}
+
+	mss_spi_write(cfg, MSS_SPI_REG_CONTROL, control);
+
+	return 0;
+}
+
+static void mss_spi_hw_release(const struct mss_spi_config *cfg)
+{
+	uint32_t control = mss_spi_read(cfg, MSS_SPI_REG_CONTROL);
+
+	control &= ~MSS_SPI_CONTROL_ENABLE;
+	mss_spi_write(cfg, MSS_SPI_REG_CONTROL, control);
+}
+
+static void mss_spi_interrupt(const struct device *dev)
+{
+	const struct mss_spi_config *cfg = dev->config;
+	struct mss_spi_data *data = dev->data;
+	struct spi_context *ctx = &data->ctx;
+	int intfield = mss_spi_read(cfg, MSS_SPI_REG_MIS) & 0xf;
+
+	if (intfield == 0) {
+		return;
+	}
+
+	/* clear the interrupt conditions */
+	if (intfield & MSS_SPI_INT_TXDONE) {
+		spi_context_complete(ctx, dev, 0);
+		mss_spi_write(cfg, MSS_SPI_REG_INT_CLEAR, MSS_SPI_INT_TXDONE);
+	}
+
+	if (intfield & MSS_SPI_INT_RXRDY) {
+		spi_context_complete(ctx, dev, 0);
+		mss_spi_write(cfg, MSS_SPI_REG_INT_CLEAR, MSS_SPI_INT_RXRDY);
+	}
+
+	if (intfield & MSS_SPI_INT_RX_CH_OVRFLW) {
+		mss_spi_write(cfg, MSS_SPI_REG_INT_CLEAR, MSS_SPI_INT_RX_CH_OVRFLW);
+		spi_context_complete(ctx, dev, 0);
+	}
+
+	if (intfield & MSS_SPI_INT_TX_CH_UNDRUN) {
+		mss_spi_write(cfg, MSS_SPI_REG_INT_CLEAR, MSS_SPI_INT_TX_CH_UNDRUN);
+		spi_context_complete(ctx, dev, 0);
+	}
+}
+
+static int mss_spi_release(const struct device *dev, const struct spi_config *config)
+{
+	const struct mss_spi_config *cfg = dev->config;
+	struct mss_spi_data *data = dev->data;
+
+	mss_spi_disable_ints(cfg);
+
+	/* release kernel resources */
+	spi_context_unlock_unconditionally(&data->ctx);
+	mss_spi_hw_release(cfg);
+
+	return 0;
+}
+
+static int mss_spi_configure(const struct device *dev, const struct spi_config *spi_cfg)
+{
+	const struct mss_spi_config *cfg = dev->config;
+	struct mss_spi_data *data = dev->data;
+	struct mss_spi_transfer *xfer = &data->xfer;
+	uint32_t control;
+
+	mss_spi_select_slave(cfg, spi_cfg->slave);
+	control = mss_spi_read(cfg, MSS_SPI_REG_CONTROL);
+	control &= ~MSS_SPI_CONTROL_ENABLE;
+	mss_spi_write(cfg, MSS_SPI_REG_CONTROL, control);
+
+	/*
+	 * Fill up the default values
+	 * Slave select behaviour set
+	 * Fifo depth greater than 4 frames
+	 * Methodology to calculate SPI Clock:
+	 *	0:	SPICLK = 1 / (2 CLK_GEN + 1) , CLK_GEN is from 0 to 15
+	 *	1:	SPICLK = 1 / (2 * (CLK_GEN + 1)) , CLK_GEN is from 0 to 255
+	 */
+
+	mss_spi_write(cfg, MSS_SPI_REG_CONTROL, xfer->control);
+
+	if (mss_spi_clk_gen_set(cfg, spi_cfg)) {
+		LOG_ERR("can't set clk divider\n");
+		return -EINVAL;
+	}
+
+	mss_spi_hw_mode_set(cfg, spi_cfg->operation);
+
+	mss_spi_write(cfg, MSS_SPI_REG_TXRXDF_SIZE, MSS_SPI_FRAMESIZE_DEFAULT);
+	control = mss_spi_read(cfg, MSS_SPI_REG_CONTROL);
+	control |= MSS_SPI_CONTROL_ENABLE;
+	mss_spi_write(cfg, MSS_SPI_REG_CONTROL, control);
+	if (spi_cfg->operation & (SPI_TRANSFER_LSB | SPI_OP_MODE_SLAVE | SPI_MODE_LOOP)) {
+		LOG_WRN("not supported operation\n\r");
+		return -ENOTSUP;
+	}
+
+	mss_spi_write(cfg, MSS_SPI_REG_COMMAND, 0xC);
```

Comment:
> It would be better to use a macro for this value '0xC'.
> 
> btw is this the enablement of the controller? If so, move it out of configure, once the buffers are setup.

Sure. I will update it.